lib/schemas.js

```diff
@@ -107,16 +107,27 @@ Schemas.prototype.getSchemas = function () {
   return Object.assign({}, this.store)
 }
 
-function buildSchemas (s) {
-  const schema = new Schemas()
-  const store = s.getSchemas()
-  Object.keys(store).forEach(schemaKey => {
-    // if the shared-schema has been used, the $id field has been removed
+Schemas.prototype.getJsonSchemas = function (options) {
+  const store = this.getSchemas()
+  const schemasArray = Object.keys(store).map(schemaKey => {
+    // if the shared-schema "replace-way" has been used, the $id field has been removed
     if (store[schemaKey]['$id'] === undefined) {
       store[schemaKey]['$id'] = schemaKey
     }
-    schema.add(store[schemaKey])
+    return store[schemaKey]
   })
+
+  if (options && options.onlyAbsoluteUri === true) {
+    // the caller want only absolute URI (without the shared schema - "replace-way" usage)
+    return schemasArray.filter(_ => !/^\w*$/g.test(_.$id))
+  }
+  return schemasArray
+}
+
+function buildSchemas (s) {
+  const schema = new Schemas()
+  const schemas = s.getJsonSchemas()
+  schemas.forEach(_ => schema.add(_))
```

Comment:
We can merge these two lines here I think to avoid allocation:

`s.getJsonSchemas().forEach(_ => schema.add(_))`